pymeasure/instruments/hp/hp8657b.py

```diff
@@ -0,0 +1,190 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2022 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import logging
+from pymeasure.instruments import Instrument
+from pymeasure.instruments.validators import strict_discrete_set, strict_range
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+
+class HP8657B(Instrument):
+    """ Represents the Hewlett Packard 8657B signal generator
+    and provides a high-level interface for interacting
+    with the instrument.
+    """
+
+    def __init__(self, adapter, **kwargs):
+        kwargs.setdefault('read_termination', '\r\n')
+        kwargs.setdefault('send_end', True)
+        super().__init__(
+            adapter,
+            "Hewlett-Packard HP8657B",
+            includeSCPI=False,
+            **kwargs,
+        )
+
+    def check_errors(self):
+        """
+        Method to read the error status register
+
+        as the 8657B does not support any readout of values, this will return 0 and log a warning
+
+        """
+        log.warning("HP8657B Does not support error status readout")
+        return 0
+
+    def clear(self):
+        """
+        Reset the instrument to power-on default settings
+
+        """
+        self.adapter.connection.clear()
+
+    id = "HP 8657B Signal generator"  #: Manual ID entry, as the instrument does not talk with us
+
+    # List of modulation sources
+    modulation_source = {"External": 1,
+                         "Int_400Hz": 2,
+                         "Int_1000Hz": 3,
+                         "OFF": 4,
+                         "DC_FM": 5}
```

Comment:
If you use an enum, the user will have autocompletion in their editor, e.g. `inst.am_source = Modulation.OFF`

Reply:
I like the idea, 
however 'Instrument.control()' does not accecpt Enums as inputs.
Or did I miss a change regarding this?

Reply:
If you use an IntEnum, the enumeration member will be interpreted as an integer, whenever an integer is expected.
I did not test it yet, but it should work, as `"Command %i" % Modulation.OFF` will be `"Command 4"`, if 
```
class Modulation(IntEnum):
    OFF = 4
```

Reply:
Thanks, it was the `map_values=True` which does not like the IntEnum.
But as this unit is listen-only we do not `map_values`